src/transactions/OfferTests.cpp

```diff
@@ -1620,4 +1688,136 @@ TEST_CASE("create offer", "[tx][offers]")
             });
         }
     }
+
+    SECTION("cannot create offer that would lead to excess liabilities")
+    {
+        SECTION("native selling liabilities")
+        {
+            for_versions_from(10, *app, [&] {
+                auto const minBalance =
+                    app->getLedgerManager().getMinBalance(2) + txfee;
+                auto acc1 = root.create("acc1", minBalance);
+                auto market = TestMarket{*app};
+
+                acc1.changeTrust(usd, 2000);
+
+                // Test when no existing offers
+                root.pay(acc1, xlm, 500 + txfee);
+                REQUIRE_THROWS_AS(
+                    market.addOffer(acc1, {xlm, usd, Price{1, 1}, 501},
```

Comment:
I think that addOffer API allows you to leave last parameter if it will be the same as second one, as it defaults to OfferState::SAME

Reply:
Good point, fixed on many lines.